src/postgres/commands/configurePostgresFirewall.ts

```diff
@@ -38,7 +38,8 @@ export async function setFirewallRule(context: IActionContext, treeItem: Postgre
     const resourceGroup: string = nonNullProp(treeItem, 'resourceGroup');
     const serverName: string = nonNullProp(treeItem, 'azureName');
 
-    const firewallRuleName: string = "azureDatabasesForVSCode-publicIp";
+    const hashCode = (s: string) => s.split('').reduce((a, b) => (((a << 5) - a) + b.charCodeAt(0)) | 0, 0);
+    const firewallRuleName: string = "azureDatabasesForVSCode-publicIp" + hashCode(ip).toString();
```

Comment:
What if you just use this:
https://github.com/microsoft/vscode-cosmosdb/blob/main/src/utils/randomUtils.ts#L9

This could be another alternative: https://github.com/microsoft/vscode-azuretools/blob/main/ui/src/utils/randomUtils.ts#L9

Both options are used pretty regularly throughout our extensions, so I prefer them over a new hash code implementation.